### DISCUSSION TITLE
random release artifact upload errors for some people, sometimes

### DISCUSSION THREAD

**Discussion by caarlos0** [community/community#14341] [Other Feature Feedback, Questions, & Ideas]
Yes, I know its a vague title, but there's a reason: this bug happens sometimes, to very few users, for a limited time, in some repositories.

It appears that uploading a file sometimes returns an error, and if the user retries it, it says the file already exists.
This seems inconsistent. How should we know if the file that is there is correct or not?

Here are a couple of cases I  collected with this behavior:

- https://github.com/cli/cli/issues/3252
- https://github.com/google/go-github/issues/2113
- https://github.com/goreleaser/goreleaser/issues/2746

As you can see, few reports, issue goes away after a while, but it definitely happens.

**Comment by blacktop** [comment]
It does seem like it is uploading the files properly, but is somehow getting an error that it didn't upload properly?  Seems that we could check if the file exists and compare CONTENT-LENGTH perhaps and then don't retry?

**Reply by caarlos0** [reply]
I think content-length is too weak of a check, checksum would probably be better... but seems like a lot of workarounds for something that should not have been happening in the first place

**Reply by blacktop** [reply]
Yeah checksum or hash would be much better, but content-length you can get with a  curl HEAD where-as with checksum wouldn't you have to pull the fill down and calculate that checksum?

**Reply by blacktop** [reply]
whoa I just tested and got weird behavior:
```bash
❯ http HEAD https://github.com/blacktop/ipsw/releases/download/untagged-01a47e6939d15870508d/ipsw_3.1.200_macOS_universal.tar.gzn Contents Restored on Nov 28, 2022 at 9:08 PM
HTTP/1.1 404 Not Found
Cache-Control: no-cache
Content-Encoding: gzip
Content-Length: 29
Content-Security-Policy: default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'
Content-Type: text/plain; charset=utf-8
Date: Tue, 29 Nov 2022 15:44:49 GMT
Referrer-Policy: no-referrer-when-downgrade
Server: GitHub.com
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
Vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-GitHub-Request-Id: C913:1A88:151E1AE:1FB2B02:638628F0
X-XSS-Protection: 0
```
a 404, but when I click on the file it downloads it no problem??

**Reply by caarlos0** [reply]
draft release I guess?
